controlplane/kubeadm/internal/webhooks/kubeadmcontrolplanetemplate_test.go

```diff
@@ -156,3 +156,153 @@ func TestKubeadmControlPlaneTemplateValidationMetadata(t *testing.T) {
 		g.Expect(warnings).To(BeEmpty())
 	})
 }
+
+func TestKubeadmControlPlaneTemplateUpdateValidation(t *testing.T) {
+	t.Run("update kubeadmcontrolplanetemplate should pass if only defaulted fields are different", func(t *testing.T) {
+		g := NewWithT(t)
+		oldKCPTemplate := &controlplanev1.KubeadmControlPlaneTemplate{
+			Spec: controlplanev1.KubeadmControlPlaneTemplateSpec{
+				Template: controlplanev1.KubeadmControlPlaneTemplateResource{
+					ObjectMeta: clusterv1.ObjectMeta{
+						Labels: map[string]string{
+							"foo":          "$invalid-key",
+							"bar":          strings.Repeat("a", 64) + "too-long-value",
+							"/invalid-key": "foo",
+						},
+						Annotations: map[string]string{
+							"/invalid-key": "foo",
+						},
+					},
```

Comment:
Nit, it is probably not relevant for the test, but what about dropping those or at least avoid invalid values

Reply:
Thx, good point. Fixed